File: CodingSamples/Java/Platform-Enterprise/FileStreamTest/met-tour/src/main/java/tourism/Site.java
package tourism;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Site implements Serializable {
    
    String title;

    private List<Visitor> visitors;

    public Site(String title) {
        this.title = title;
        visitors = new ArrayList<>();
    }

    public String getTitle() {
        return title;
    }

    public List<Visitor> getVisitors() {
        return visitors;
    }

    public Visitor getVisitor(String name) {
        return visitors.stream()
                .filter(v -> v.name.equals(name))
                .findFirst()
                .orElseGet(() -> {
                    var visitor = new Visitor(name);
                    visitors.add(visitor);
                    return visitor;
                });
    }

    public void persist() {
        SiteStore.save(this);
    }
}

File: CodingSamples/DotNet/Platform-Enterprise/FileStreamTest/MetTourLib/Tourism/Site.cs
namespace Tourism;

public class Site(string title)
{
    public string Title { get; set; } = title;

    public List<Visitor> Visitors { get; set; } = [];

    public Visitor GetVisitor(string id)
    {
        var visitor = Visitors.Find(v => v.Name == id);
        if(visitor is null)
        {
            visitor = new Visitor { Name = id };
            Visitors.Add(visitor);
        }
        return visitor;
    }
}